lib/clerk/rack_middleware_v2.rb

```diff
@@ -170,8 +170,19 @@ def call(env)
         return unknown(interstitial: true)
       end
 
-      if production? && client_uat.nil?
-        return signed_out(env)
+      # Show interstitial when there is no client_uat or cookie token
+      if (client_uat == "0" || client_uat.nil?) && cookie_token.to_s.empty?
+        return unknown(interstitial: true)
+      end
+
+      # Show interstitial when there is client_uat is incompatible with cookie token
+      has_cookie_token_without_client = (client_uat == "0" || client_uat.nil?) && cookie_token
+      has_client_without_cookie_token = (client_uat.to_s != "0" || client_uat.to_s != "") && cookie_token.to_s.empty?
```

Comment:
❓ Where does the `client_uat.to_s != ""` case correspond here to?

Reply:
it corresponds to the case where the `client_uat` is missing or the value is somehow removed. We could use the `client_uat.nil?` if that makes it more readable but i think this check is safer because it also applies to cases where the client_uat exists without value (due to some edge-case or extension or something).